types/pixi/placeableObjects/token.d.ts

```diff
@@ -1,88 +1,133 @@
 /**
- * An instance of the Token class represents an Actor within a viewed Scene on the game canvas.
- * Each Token is reference using a numeric id which indexes its position within the scene.
- * See the initialization signature of the parent PlaceableObject class for more details.
+ * A Token is an implementation of PlaceableObject which represents an Actor within a viewed Scene on the game canvas.
  *
- * @param data - An object of token data which is used to construct a new Token.
- * @param scene - The parent Scene entity within which the Token resides.
+ * @example
+ * ```typescript
+ * Token.create<Token>({
+ *   name: "Token Name",
+ *   x: 1000,
+ *   y: 1000,
+ *   displayName: 3,
+ *   img: "path/to/token-artwork.png",
+ *   width: 2,
+ *   height: 2,
+ *   scale: 1.2,
+ *   elevation: 50,
+ *   lockRotation: false,
+ *   rotation: 30,
+ *   effects: ["icons/stun.png"],
+ *   overlayEffect: "icons/dead.png",
+ *   vision: true,
+ *   dimSight: 60,
+ *   brightSight: 0,
+ *   dimLight: 40,
+ *   brightLight: 20,
+ *   sightAngle: 60,
+ *   hidden: false,
+ *   actorId: "dfgkjt43jkvdfkj34t",
+ *   actorLink: true,
+ *   actorData: {},
+ *   disposition: 1,
+ *   displayBars: 3,
+ *   bar1: {attribute: "attributes.hp"},
+ *   bar2: {attribute: "attributes.sp"}
+ * }
+ * ```
  */
-declare class Token extends PlaceableObject {
+declare class Token extends PlaceableObject<Token.Data> {
   /**
-   * An Actor entity constructed using this Token's data
-   * If actorLink is true, then the entity is the true Actor entity
-   * Otherwise, the Actor entity is a synthetic, constructed using the Token actorData
+   * A Ray which represents the Token's current movement path
    */
-  actor: Actor;
+  protected _movement: Ray | null;
 
   /**
-   * Convenience access to the token's nameplate string
+   * An Object which records the Token's prior velocity dx and dy
+   * This can be used to determine which direction a Token was previously moving
    */
-  name: string;
+  protected _velocity: {
+    dx: number | null;
+    dy: number | null;
+    sx: number | null;
+    sy: number | null;
+  };
+
+  /**
+   * The Token's most recent valid position
+   */
+  protected _validPosition: { x: number; y: number };
+
+  /**
+   * Provide a temporary flag through which th6is Token can be overridden to bypass any movement animation
+   */
+  _noAnimate: boolean;
 
   /**
    * Track the set of User entities which are currently targeting this Token
    */
   targeted: Set<User>;
 
   /**
-   * A Ray which represents the Token's current movement path
+   * An Actor entity constructed using this Token's data
+   * If actorLink is true, then the entity is the true Actor entity
+   * Otherwise, the Actor entity is a synthetic, constructed using the Token actorData
+   * @remarks
+   * This should be typecast to your systems actor if needed.
    */
-  protected _movement: any;
+  actor: Actor;
```

Comment:
Thoughts on handling typing here? Leave as is or add a generic to the class for Actor type?

Reply:
I suppose with the current state of things, adding a type parameter for `Actor` would be the proper thing. I have no clue how practical that is though, depending on the amount of places that we would need to add it to (and if it's available in those places).

We have the same problem with `Actor` and `Item` etc. already, e.g. currently it's not possible to have `game.actors` be of the correct type, because the `Game` class does not know anything about specific `Actor` types. And don't know of a way to do that without using global overrides (which @kmoschcau has some valid arguments against).

Reply:
Yeah with the singleton `game` instance, it's a pickle. Would it be possible to give `Game` generic params for the `Actor` and have some way for the user to say that `game` is a `Game` instance with their custom `Actor`s? The generic params would just default to the normal `Actor` class.

Reply:
The problem with that approach is that `Game` is declared as a class. It would need to be declared as a type in order for the User to be able to redeclare it (which is possible I guess).

Reply:
Well in that case if we want to be strict, the only thing we can do, is use the default `Actor` and the user then has to check.